modules/map/tools/map_datachecker/client/client.cc

```diff
@@ -0,0 +1,250 @@
+        << channel_checker_stop_flag_file_;
+}
+
+int Client::run() {
+  std::string stage = FLAGS_stage;
+  AINFO << "stage [" << stage << "]";
+  if ("record_check" == stage) {
+    record_check();
+  } else if ("static_align" == stage) {
+    static_align();
+  } else if ("eight_route" == stage) {
+    eight_route();
+  } else if ("data_collect" == stage) {
+    data_collect();
+  } else if ("loops_check" == stage) {
+    loops_check();
+  } else if ("clean" == stage) {
+    clean();
+  }
+  return 0;
+}
+
+int Client::record_check() {
+  std::string cmd = FLAGS_cmd;
+  ChannelChecker channel_checker(channel_checker_stop_flag_file_);
+  AINFO << "cmd [" << cmd << "]";
+  if ("start" == cmd) {
+    std::string record_path = FLAGS_record_path;
+    AINFO << "record_path [" << record_path << "]";
+    if (!boost::filesystem::exists(record_path)) {
+      AERROR << "record_path does not exist";
+      return -1;
+    }
+    int ret = 0;
+    ret = channel_checker.sync_start(record_path);
+    if (ret != 0) {
+      AERROR << "sync_start channel chacker failed, record_path ["
+             << record_path << "]";
+      return -1;
+    }
+  } else {
+    int ret = 0;
+    ret = channel_checker.sync_stop();
+    if (ret != 0) {
+      AERROR << "sync_stop channel chacker failed";
+      return -1;
+    }
+  }
+  return 0;
+}
+
+int Client::static_align() {
+  std::string cmd = FLAGS_cmd;
+  AINFO << "cmd [" << cmd << "]";
+  StaticAlign static_align;
+  if ("start" == cmd) {
+    int ret = 0;
+    ret = static_align.sync_start();
+    if (ret != 0) {
+      AERROR << "sync_start static align failed";
+      return -1;
+    } else {
+      AINFO << "Static aligh succeed";
+      fprintf(USER_STREAM,
+              "Static aligh succeed. Next, you may want to run: bash client.sh "
+              "-- stage eight_route\n");
+    }
+  } else {
+    int ret = 0;
+    ret = static_align.sync_stop();
+    if (ret != 0) {
+      AERROR << "sync_stop static align failed";
+      return -1;
+    }
+  }
+  return 0;
+}
+
+int Client::eight_route() {
+  std::string cmd = FLAGS_cmd;
+  AINFO << "cmd [" << cmd << "]";
+  EightRoute eight_route;
+  if ("start" == cmd) {
+    int ret = 0;
+    ret = eight_route.sync_start();
+    if (ret != 0) {
+      AERROR << "sync_start static align failed";
+      return -1;
+    } else {
+      AINFO << "Eight route succeed";
+      fprintf(USER_STREAM,
+              "Eight route succeed. Next, you may want to run: bash client.sh "
+              "--stage data_collect\n");
+    }
+  } else {
+    int ret = 0;
+    ret = eight_route.sync_stop();
+    if (ret != 0) {
+      AERROR << "sync_stop static align failed";
+      return -1;
+    }
+  }
+  return 0;
+}
+
+int Client::data_collect() {
+  std::string cmd = FLAGS_cmd;
+  AINFO << "cmd [" << cmd << "]";
+  std::vector<std::string> lines = get_file_lines(data_collect_time_flag_file_);
+  std::ofstream time_file_handler(data_collect_time_flag_file_);
```

Comment:
Even though it may be rare, it's better to check whether the file is successfully opened. For example, if you don't have permission to create a file.